<launch>

<!--Parameters for Kinematics -->
    <arg name="frontLength"         default="-0.383"/>
    <arg name="rearLength"          default="-0.383"/>
    <arg name="wheelDiameter"       default="0.280"/>
    <arg name="axesLength"          default="0.335"/>


    <node name="base" pkg="base" type="base_node">
        <param name="wheelDiameter"     value="$(arg wheelDiameter)"/>
        <param name="axesLength"        value="$(arg axesLength)"/>

        <!--remap from="/engine/actualSpeed"            to="/Engine/ActualSpeed"/-->
        <!--remap from="/engine/targetTorque"           to=""/-->
        <!--remap from="/engine/targetSpeed"            to="/Engine/TargetSpeed"/-->
        <!--remap from="/engine/targetAcceleration"     to=""/-->
    </node>

   
    <!--args="tx ty tz rx ry rz rw" Translation kartesisch Rotation quaternion-->
    <node pkg="tf2_ros" type="static_transform_publisher" name="axesFront2jointFront"   args="$(arg frontLength) 0 0 0 0 0 1 axesFront jointFront" />
    <node pkg="tf2_ros" type="static_transform_publisher" name="jointRear2axesRear"     args="$(arg rearLength) 0 0 0 0 0 1  jointRear axesRear" />
    <node pkg="tf2_ros" type="static_transform_publisher" name="base_link2axesFront"                  args="0 0 0 0 0 0 1 base_link axesFront" />

    <!--include file="$(find sick_scan)/launch/own_scanner.launch">
        <arg name="ParentFrame" value="AxesFront" />
        <arg name="PositionID"  value="Front" />
        <arg name="TX"          value="0" />
        <arg name="TY"          value="0" />
        <arg name="TZ"          value="-0.1" />
        <arg name="IPAddress"   value="192.168.0.51" />
    </include>

    <include file="$(find sick_scan)/launch/own_scanner.launch">
        <arg name="ParentFrame" value="AxesRear" />
        <arg name="PositionID"  value="Rear" />
        <arg name="TX"          value="0" />
        <arg name="TY"          value="0" />
        <arg name="TZ"          value="-0.1" />
        <arg name="IPAddress"   value="192.168.0.52" />
    </include-->

</launch>
